cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java

```diff
@@ -0,0 +1,137 @@
+    this.tetheredInstanceName = conf.getTetheredInstanceName();
+    this.tetheredNamespace = conf.getTetheredNamespace();
+    this.cConf = cConf;
+    this.messagePublisher = new MultiThreadMessagingContext(messagingService).getMessagePublisher();
+  }
+
+  @Override
+  public void launch(RuntimeJobInfo runtimeJobInfo) throws IOException {
+    ProgramRunInfo runInfo = runtimeJobInfo.getProgramRunInfo();
+    LOG.debug("Launching run {} with following configurations: tethered instance name {}, tethered namespace {}.",
+              runInfo.getRun(), tetheredInstanceName, tetheredNamespace);
+    byte[] payload = Bytes.toBytes(GSON.toJson(createPayload()));
+    TetheringControlMessage message = new TetheringControlMessage(TetheringControlMessage.Type.RUN_PIPELINE, payload);
+    publishToControlChannel(message);
+  }
+
+  @Override
+  public Optional<RuntimeJobDetail> getDetail(ProgramRunInfo programRunInfo) {
+    // TODO: pull job status instead of always treating it as RUNNING
+    return Optional.of(new RuntimeJobDetail(programRunInfo, RuntimeJobStatus.RUNNING));
+  }
+
+  @Override
+  public List<RuntimeJobDetail> list() throws Exception {
+    // TODO: pull list of all running jobs in tethered instance. This method is unused
+    return new ArrayList<>();
+  }
+
+  @Override
+  public void stop(ProgramRunInfo programRunInfo) throws Exception {
+    RuntimeJobDetail jobDetail = getDetail(programRunInfo).orElse(null);
+    if (jobDetail == null) {
+      return;
+    }
+    RuntimeJobStatus status = jobDetail.getStatus();
+    if (status.isTerminated() || status == RuntimeJobStatus.STOPPING) {
+      return;
+    }
+    LOG.debug("Stopping run {} with following configurations: tethered instance name {}, tethered namespace {}.",
+              programRunInfo.getRun(), tetheredInstanceName, tetheredNamespace);
+    byte[] payload = Bytes.toBytes(GSON.toJson(programRunInfo));
+    TetheringControlMessage message = new TetheringControlMessage(TetheringControlMessage.Type.STOP_PIPELINE, payload);
+    publishToControlChannel(message);
+  }
+
+  @Override
+  public void kill(ProgramRunInfo programRunInfo) throws Exception {
+    stop(programRunInfo);
+  }
+
+  @Override
+  public void close() {
+    // no-op
+  }
+
+  // TODO: create payload containing necessary files
+  private Map<String, String> createPayload() {
+    return null;
+  }
+
+  @VisibleForTesting
+  void publishToControlChannel(TetheringControlMessage message) throws IOException {
+    TopicId topicId = new TopicId(NamespaceId.SYSTEM.getNamespace(),
+                                  cConf.get(Constants.Tethering.TOPIC_PREFIX) + tetheredInstanceName);
+    try {
+      messagePublisher.publish(topicId.getNamespace(), topicId.getTopic(), StandardCharsets.UTF_8,
+                               GSON.toJson(message));
+    } catch (IOException | TopicNotFoundException e) {
+      LOG.error("Failed to publish to topic {}", topicId, e);
+      throw new IOException(e);
```

Comment:
Do we need to wrap the exception in an IOException? Can't we just throw whatever exception was raised?